lib/src/react_client/synthetic_event_wrappers.dart

```diff
@@ -50,26 +64,35 @@ class SyntheticKeyboardEvent extends SyntheticEvent {
 }
 
 @JS()
-@anonymous
 class SyntheticCompositionEvent extends SyntheticEvent {
+  /// [SyntheticCompositionEvent]s cannot be manually instantiated.
+  ///
+  /// Use `createSyntheticCompositionEvent` instead.
+  external factory SyntheticCompositionEvent._();
+
   external String get data;
 }
 
 @JS()
-@anonymous
 class SyntheticFocusEvent extends SyntheticEvent {
+  /// [SyntheticFocusEvent]s cannot be manually instantiated.
+  ///
+  /// Use `createSyntheticFocusEvent` instead.
+  external factory SyntheticFocusEvent._();
+
   external EventTarget get relatedTarget;
 }
 
 @JS()
-@anonymous
-class SyntheticFormEvent extends SyntheticEvent {}
+class SyntheticFormEvent extends SyntheticEvent {
+  /// [SyntheticFormEvent]s cannot be manually instantiated.
+  ///
+  /// Use `createSyntheticFormEvent` instead.
+  external factory SyntheticFormEvent._();
+}
 
 /// A JS object that looks like a [DataTransfer] but isn't one.
-///
-/// See `syntheticDataTransferFactory` for more info.
 @JS()
-@anonymous
 class NonNativeDataTransfer {
```

Comment:
I excluded this because it seemed like a special case since it isn't actually an event, but let me know if it should be treated like one and receive a `createNonNativeDataTransfer` API and have it's constructor overridden.

Reply:
Crap, I forgot about this... 

tl;dr sometimes the `dataTransfer` property we get from React is a native `DataTransfer` object, and sometimes it's not, and mistyping it either way has caused issues in the past.

So, below, I don't think we can type `SyntheticMouseEvent.dataTransfer` as `NonNativeDataTransfer`.

See https://github.com/cleandart/react-dart/pull/151, https://github.com/cleandart/react-dart/pull/154, and https://github.com/cleandart/react-dart/pull/220 for more background info

I don't think we ever discovered or looked into when and why the non-native datatransfer object got in there... I talked to Evan and he couldn't remember exactly where he was hitting it:
> evan.weible
> I think it was when I was messing around with drag-and-drop file uploads
> …but I can’t remember much more than that 😕 
> 
> greg.littlefield
> Oooh nice, nice. Do you remember if it was a thing that happened when you were interacting with the app vs hitting it in tests?
>
> evan.weible
> I think when interacting with the app
> but really not sure

Some theories:
- It's from React internals passing in a non-native event somewhere
    - I couldn't find evidence of this; React seems to just pass it through
- It's from test utilities that pass in a JS object stub of a datatransfer object: https://sourcegraph.wk-dev.wdesk.org/search?q=%5B%27%22%5DdataTransfer%5B%27%22%5D+lang:dart++count:1000&patternType=regexp
- The property is being polyfilled in browsers or replaced by an extension

I hate to go down this rabbit hole, but now is probably the time to get to the bottom of this, since we no longer have a layer ([syntheticDataTransferFactory](https://github.com/cleandart/react-dart/pull/284/files#diff-00a14801d8125575e2b4e7ad2eb7b74ae9c57162f484a8d4e7f792416382a9b0L127-L128)) in which we can handle both.

....Oooh, unless we replaced the `dataTransfer` getter with an extension getter that continues to return our `SyntheticDataTransfer` object...

That might be a better solution for now, since I'd imagine chasing it down the root issue is going to be a PITA.

---

Sorry, that was a lot haha. Thoughts?

Reply:
Oof that's a fun one haha. I agree it's probably best to use an extension to keep the current functionality. Part of me does really want to dig in and figure out what's happening, and I agree it's a good time.

Part of my logic for procrastinating it a moment more is that with all the event changes, WSD is already in a weird state. I'd want to use WSD to test the fix for `dataTransfer` though, so if that were part of this ticket it'd make getting WSD fixed a dependency to this. Whereas we can do that a little more piecemeal if we do this change, then fix WSD, and then come back to `dataTransfer`. How would you feel about using the extension for now and discussing during a standup or planning to see if it's worth digging into the real cause before the 6.0.0 release? Or would you rather we just do it now if we ever do?

Reply:
@greglittlefield-wf forgot to tag ya haha

Reply:
I'd personally be fine with the extension for the 6.0.0 release, since there's not really any benefit to actually fixing the issue as opposed to continuing to use the `SyntheticDataTransfer`

Reply:
But Greg, isn't your inner engineer just dying to know how that's getting mutated? 😉 

That's fair though, it makes sense we don't need to dig into it until we have reason to believe it's causing an issue. So I will make that change and be good to go!

Reply:
I added a helper in `lib/src/react_client/event_helpers.dart` and deprecated the `dataTransfer` field on the interop class to try to achieve this.. let me know if this wasn't what you were thinking though.

If that is the best path, I could go back into `react.dart` and add the helper there as well so we can still forward fix

Reply:
I just added a comment on the extension

Reply:
Doope, my interop skills are still lacking and I forget about those methods 🤦 Thank you!!